docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,646 @@ under the License.
 
-A Sub-Task is a [Task](#task) responsible for processing a [partition](#partition) of
-the data stream. The term "Sub-Task" emphasizes that there are multiple parallel Tasks for the same
-[Operator](#operator) or [Operator Chain](#operator-chain).
+#### Stream
+
+This is a concept describing a continuous flow of data (generated by various sources) that is fed into 
+stream processing systems to be processed and analyzed in real time. It consists of a series of data 
+elements ordered in time. 
+
+[DataStream](#flink-datastream) is the representation of this concept inside Flink's DataStream API 
+and Table API.
+
+#### Stream Barriers
+
+A core element of Flink's distributed snapshotting. Stream barriers are injected into the [DataStream](#datastream) 
+and flow with the [records](#record) as part of the DataStream. Barriers never overtake records and 
+flow strictly in line. A barrier separates the records in the DataStream into the set of records that 
+goes into the current snapshot, and the records that go into the next snapshot.
+
+#### Subtask
+
+A subtask is an abstract concept that describes a [task](#task) responsible for processing a [partition](#partition) 
+of the [DataStream](#datastream). The term "subtask" emphasizes that there are multiple parallel tasks 
+for the same [operator](#operator) or [operator chain](#operator-chain).
 
 #### Table Program
 
 A generic term for pipelines declared with Flink's relational APIs (Table API or SQL).
 
 #### Task
 
-Node of a [Physical Graph](#physical-graph). A task is the basic unit of work, which is executed by
-Flink's runtime. Tasks encapsulate exactly one parallel instance of an
-[Operator](#operator) or [Operator Chain](#operator-chain).
+This is a node in a [physical graph](#physical-graph). A task is the basic unit of work which is executed 
+by Flink's runtime. Tasks encapsulate exactly one parallel instance of an [operator](#operator) or
+[operator chain](#operator-chain).
+
+#### Task Chaining
+
+This is an optimization where Flink puts two subsequent [transformations](#transformation) in the same thread, if possible. 
+
+#### Task Parallelism
+
+This is the number of parallel instances of a task. A [Flink application](#flink-application) consists 
+of multiple [tasks](#task) ([transformations](#transformation), [operators](#operator), [sources](#source), 
+[sinks](#sink)). A task is split into several parallel instances for execution and each parallel instance 
+processes a subset of the task's input data. 
 
-#### Flink TaskManager
+#### Task Slot
 
-TaskManagers are the worker processes of a [Flink Cluster](#flink-cluster). [Tasks](#task) are
-scheduled to TaskManagers for execution. They communicate with each other to exchange data between
-subsequent Tasks.
+This is one unit of resource scheduling in a [Flink cluster](#(flink)-cluster). Each task slot 
+represents a fixed subset of resources of the [TaskManager](#taskmanager). The number of task slots 
```

Comment:
This isn't correct. All of the task slots in a TM are competing with each other for the resources available to the JVM that is running the TM. Nothing constrains a slot to only consume a fixed subset of those resources.